### PR TITLE
rgw: bucket stats display bucket index type

### PR DESCRIPTION
**Pull request by fangyuxiangGL**
Signed-off-by: fang yuxiang fang.yuxiang@eisoo.com

### CODE REVIEW

**Comment by fangyuxiangGL** [comment]
@Liuchang0812 @cbodley 

hi, optimize the codes as your suggestion, but i made something wrong that cause the previous branch was deleted.

could you review again for me? thx

File: src/rgw/rgw_common.h [hunk 1]
```diff
@@ -1102,6 +1102,28 @@ enum RGWBucketIndexType {
...
+      str = "Normal";
+      break;
+    }
+    case RGWBIType_Indexless:
+    {
+      str = "Indexless";
+      break;
+    }
+    default:
+      break;
```
Comment (Liuchang0812) [inline src/rgw/rgw_common.h]: how about 'str = "Unknown"?

Comment (fangyuxiangGL) [inline src/rgw/rgw_common.h]: ok. :+1:

**Comment by fangyuxiangGL** [comment]
retest this please

File: src/rgw/rgw_common.h [hunk 2]
```diff
@@ -1102,6 +1102,31 @@ enum RGWBucketIndexType {
...
+    default:
+    {
+      str = "Unknown";
+      break;
+    }
+  }
+  
+  return out << str;
+}
+
```
Comment (cbodley) [inline src/rgw/rgw_common.h]: can you try to shorten this function a bit? the temporary `string str` shouldn't be necessary. something like:
```c++
inline std::ostream& operator<<(std::ostream& out, RGWBucketIndexType index_type)
{
  switch (index_type) {
  case RGWBIType_Normal:
    return out << "Normal";
  case RGWBIType_Indexless:
    return out << "Indexless";
  default:
    return out << "Unknown";
  }
}
```

Comment (fangyuxiangGL) [inline src/rgw/rgw_common.h]: ok~ done ..

**Pull request merged**